### PR TITLE
AndroidX

### PR DESCRIPTION
**Pull request by caiofaustino**
## Summary
- Update to AndroidX libraries
- Update other old dependencies.
- Add Kotlin to all modules.
- Bump min API to 21

**Fixed issue**: 
#217

### CODE REVIEW

File: drop-in/src/main/java/com/adyen/checkout/dropin/service/DropInService.kt [hunk 1]
```diff
@@ -11,13 +11,15 @@ package com.adyen.checkout.dropin.service
...
-import android.support.v4.content.LocalBroadcastManager
+import androidx.core.app.JobIntentService
 import com.adyen.checkout.base.model.payments.request.PaymentComponentData
 import com.adyen.checkout.base.model.payments.request.PaymentMethodDetails
 import com.adyen.checkout.core.exception.CheckoutException
 import com.adyen.checkout.core.log.LogUtil
 import com.adyen.checkout.core.log.Logger
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.callbackFlow
```
Comment (jrodbx) [inline drop-in/src/main/java/com/adyen/checkout/dropin/service/DropInService.kt:22]: out of curiosity, would it be easier to land this androidx migration without the coroutines migration?

File: drop-in/src/main/java/com/adyen/checkout/dropin/ui/paymentmethods/PaymentMethodListDialogFragment.kt [hunk 2]
```diff
@@ -73,7 +73,7 @@ class PaymentMethodListDialogFragment : DropInBottomSheetDialogFragment(), Payme
                         ImageLoader.getInstance(requireContext(), mDropInViewModel.dropInConfiguration.environment),
                         arguments?.getBoolean(SHOW_IN_EXPAND_STATUS)!!,
                         this)
-                recyclerView.layoutManager = LinearLayoutManager(requireContext())
+                recyclerView.layoutManager = androidx.recyclerview.widget.LinearLayoutManager(requireContext())
```
Comment (jrodbx) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ui/paymentmethods/PaymentMethodListDialogFragment.kt:76]: nit: imports?  (same in a few other source files)

**Review by jrodbx — COMMENTED** [review]
Looks great.  Hoping to see this land soon!

**Comment by caiofaustino** [comment]
Thanks @jrodbx, we are finishing up an internal security process to move our development flow here to GitHub, took a bit longer than I expected, but it's coming along now. After that I'll branch out and release this as an `alpha` version for you guys to test and start using if you want while we might still change some other things before a formal release. Do you think that's OK?

**Comment by jrodbx** [comment]
SGTM!

File: drop-in/src/main/java/com/adyen/checkout/dropin/ui/paymentmethods/PaymentMethodListDialogFragment.kt (hunk 2 above)
Comment (caiofaustino) [reply drop-in/src/main/java/com/adyen/checkout/dropin/ui/paymentmethods/PaymentMethodListDialogFragment.kt:76]: Thx for pointing this out, I will fix this on another PR.

**Pull request merged**